src/index.ts

```diff
@@ -51,6 +57,9 @@ export const beautifier: Beautifier = {
   }: BeautifierBeautifyData) {
     return new Promise<string>((resolve, reject) => {
       const { CLIEngine } = dependencies.get<NodeDependency>("ESLint").package;
+      const ESLintPluginReact = dependencies.get<NodeDependency>(
```

Comment:
To allow for the React plugin to be optional, we should check `beautifierConfig` (or `config` after line 63).

Installation instructions from https://www.npmjs.com/package/eslint-plugin-react show:

```
 "extends": [
    "eslint:recommended",
    "plugin:react/recommended"
  ]
```

So we should check for `plugin:react/` and if it exists in `extends` then get the React plugin dependency (line 60) and add the plugin (line 79). 

cc @szeck87 for https://github.com/Unibeautify/beautifier-eslint/issues/87#issuecomment-417022796

Reply:
You can try that.  If that can't be done, `dependencies.get<NodeDependency>("ESLint-plugin-React")` will have an error that you can check for.

Reply:
Whoops. @szeck87 pointed out `.has` will not work here.

We actually want to use `dependency.isInstalled`: https://api.unibeautify.com/classes/_dependencymanager_dependency_.dependency.html#isinstalled

So more like this:

```typescript
const reactPlugin = dependencies.get<NodeDependency>("ESLint-plugin-React");
if (reactPlugin.isInstalled) {
  cli.addPlugin("eslint-plugin-react", reactPlugin.package);
}
```

Feel free to clean up the above.

Reply:
Yea I noticed that too, might have to comment on the line above the `if` statement:

`// tslint:disable-next-line:promise-must-complete`

Reply:
Where is this error coming from @muuvmuuv ? Could you push a commit with this?

Maybe you already did push a commit. I currently see Travis CI failing:

```
npm ERR! Unexpected token } in JSON at position 994
```